src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -220,6 +220,7 @@ function VisEditor($scope, $route, timefilter, AppState, $window, kbnUrl, courie
 
     const updateTimeRange = () => {
       $scope.timeRange = timefilter.time;
+      $scope.vis.forceReload();
```

Comment:
As discussed we should instead listen in visualize on the changes of filter/queries/timerange